.github/wiki/Tests.md

```diff
@@ -0,0 +1,135 @@
+# Tests
+
+## Running Tests
+This section is about testing your changes to React Native as a contributor. If you haven't yet, go through the steps to set up your development environment for [building projects with native code][env-setup]. If you are intested in [writing tests](#writing-tests) for a React Native app, proceed to the next section.
+
+[env-setup]: http://facebook.github.io/react-native/docs/getting-started
+
+
+### JavaScript Tests
+
+The simplest way to run the JavaScript test suite is by using the following command at the root of your React Native checkout:
+
+```bash
+npm test
+```
+
+This will run tests using [Jest](https://jestjs.io).
+
+You should also make sure your code passes [Flow](https://flowtype.org/) and lint tests:
+
+```bash
+npm run flow
+npm run lint
+```
+
+### iOS Tests
+
+To run the iOS tests, invoke the following script from the root of your React Native checkout:
+
+```bash
+./scripts/objc-test-ios.sh test
```

Comment:
Should these have npm scripts?

Reply:
They don't at this time, but we should set it up. There's a lot of cleanup that can be done here, for example, there's three files: objc-test-ios.sh, objc-test-tvos.sh, and objc-test.sh. The latter should not be invoked directly, as the first two just set up the necessary envvars.

I actually wrote most of this doc over three months ago but have deferred adding it as I kept encountering areas where the DX could be improved, so I am leaning towards documenting how things are today even if its painful in some areas. Getting more eyes on it may help us get an idea which of these we should focus on fixing first.

Reply:
Ah yeah totally agree